### DISCUSSION TITLE
Is threre any detail explanations about algorithm used in Manybody Calculation?

### DISCUSSION THREAD

**Discussion by loadingcn** [cosmograph-org/cosmos#83] [Q&A] [answered]
Hi there, I'm trying to improve cosmo run ten million nodes by simplify some code.
But the ForceManyBody part seems a little bit confusing to me.
As I see, this.calculateLevelsCommand  makes a quad-tree as Barnes-Hut algorithm does. 
But I can't understand this.forceCommand.
Appreciate if someone can help!

**Comment by rokotyan** [comment, marked as answer]
In short, this command searches for the coordinates of the "cells" at each level of detail from which nodes should repel. Each "cell" stores the center of mass and the number of particles it contains. The size of the cell depends on the level.

**Reply by loadingcn** [reply]
Thanks a lot!!!

**Reply by loadingcn** [reply]
In the process of calulating the "cell", what does the "theta" represent for? 

n_left = max(0.0, floor(dist_left / cellSize - theta))

It's quite big (default 1.15) compared to the division result sometimes.

**Reply by rokotyan** [reply]
@loadingcn Check out this Codepen https://codepen.io/stukova/pen/wvyajgV, it shows how the algorithm calculates the cells.

**Reply by loadingcn** [reply]
Thanks!!!
